scripts/write_deployinst_input.py

```diff
@@ -188,66 +187,124 @@ def read_template(template):
+                   'assem_size': int(8000 / 473),
+                   'n_assem_core': 473 / 500,
+                   'n_assem_batch': 60}
+    pwr_spec = {'template' : pwr_template,
+                   'assem_size': 446.0,
+                   'n_assem_core': 193 / 1000,
+                   'n_assem_batch': 193 / 3000}
+    epr_spec = {'template' : pwr_template,
+                   'assem_size': 467.0,
+                   'n_assem_core': 216,
+                   'n_assem_batch': 72}
+
+    reactor_specs = {'AP1000' : ap1000_spec,
+                    'PHWR': phwr_spec,
+                    'BWR': bwr_spec,
+                    'CANDU': candu_spec,
+                    'PWR': pwr_spec,
+                    'EPR': epr_spec}
+
+    for data in data_list:
+        # refine name string
+        name = refine_name(data['reactor_name'])
+        reactor_type = data['type'].decode('utf-8')
+        if reactor_type in reactor_specs.keys():
+            # if the reactor type matches with the pre-defined dictionary,
+            # use the specifications in the dictionary.
+            reactor_str = reactor_type
+            spec_dict = reactor_specs[reactor_str]
+            reactor_body = spec_dict['template'].render(
                 country=data['country'].decode('utf-8'),
+                type=reactor_str,
                 reactor_name=name,
-                assem_size=180,
-                n_assem_core=int(round(data['capacity']/1000 * 764)),
-                n_assem_batch=int(round(data['capacity']/3000 * 764)),
+                assem_size=spec_dict['assem_size'],
+                n_assem_core=int(round(spec_dict['n_assem_core'] * data['capacity'])),
+                n_assem_batch=int(round(spec_dict['n_assem_batch'] * data['capacity'])),
                 capacity=data['capacity'])
-        # if French PWR, use mox template for mox reactor
-        elif data['type'].decode('utf-8') == 'PWR' and data['country'].decode('utf-8') == 'France':
-            reactor_body = mox_template.render(country=data['country'].decode('utf-8'),
-                                               reactor_name=name,
-                                               assem_size=523.4,
-                                               n_assem_core=int(
-                                                   round(data['capacity']/1000 * 193)),
-                                               n_assem_batch=int(
-                                                   round(data['capacity']/3000 * 193)),
-                                               capacity=data['capacity'])
-        # if not BWRS, all go with PWR specification.
         else:
-            reactor_body = template.render(
+            # assume 1000MWe pwr linear core size model if no match
+            reactor_body = pwr_template.render(
                 country=data['country'].decode('utf-8'),
                 reactor_name=name,
+                type=data['type'].decode('utf-8'),
```

Comment:
I think this variable has already been given a useful name above (reactor_type = data['type'].decode('utf-8')) so you should use that name here rather than retrieving and decoding it again.